pkg/querier/querier.go

```diff
@@ -553,74 +551,6 @@ func (querier) Close() error {
 	return nil
 }
 
-func (q querier) mergeSeriesSets(sets []storage.SeriesSet) storage.SeriesSet {
-	// Here we deal with sets that are based on chunks and build single set from them.
-	// Remaining sets are merged with chunks-based one using storage.NewMergeSeriesSet
-
-	otherSets := []storage.SeriesSet(nil)
-	chunks := []chunk.Chunk(nil)
-
-	for _, set := range sets {
-		nonChunkSeries := []storage.Series(nil)
-
-		// SeriesSet may have some series backed up by chunks, and some not.
-		for set.Next() {
-			s := set.At()
-
-			if sc, ok := s.(SeriesWithChunks); ok {
```

Comment:
I think we use `SeriesWithChunks` only in `streamingSelect` when querying Ingesters and the merge already happened there.

I am a little bit unsure why we need to get those chunks again and partition them again into a series set using `partitionChunks`. Isn't it already a series set? We already only get 1 input series set with chunks since we only have 1 distributor querier.